src/scripts/main.js

```diff
@@ -1,3 +1,403 @@
+  loseMessage.classList = ['message message-lose'];
+  startMessage.classList = ['message message-start hidden'];
+}
+
+function noEmptyCell() {
+  for (const row of table) {
+    for (const cell of row) {
+      if (cell === 0) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function noCellToMerge() {
+  for (const row of table) {
+    for (let i = 0; i < row.length - 1; i++) {
+      const current = row[i];
+      const next = row[i + 1];
+
+      if (current === next) {
+        return false;
+      }
+    }
+  }
+
+  for (let j = 0; j < 3; j++) {
+    for (let k = 0; k < 3; k++) {
+      const current = table[k][j];
+      const cellNextRow = table[k + 1][j];
+
+      if (current === cellNextRow) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function loseCheck() {
+  if (noEmptyCell() && noCellToMerge()) {
+    messageLose();
+  }
+}
+
+function updateScore() {
+  score.textContent = scoreValue;
+
+  for (const row of table) {
+    for (const item of row) {
+      if (item === 2048) {
+        messageWin();
+      }
+    }
+  }
+}
+
+let scoreValue = 0;
+const score = document.querySelector('.game-score');
+
+score.textContent = scoreValue;
+
+const fields = document.querySelectorAll('.field-cell');
+const button = document.querySelector('.button');
+
+function updateGameFields() {
+  for (let i = 0; i < fields.length; i++) {
+    const indexRow = Math.floor(i / 4);
+    const indexCol = i - (indexRow * 4);
+    const tableValue = table[indexRow][indexCol];
+
+    fields[i].textContent = tableValue;
+
+    fields[i].classList = ['field-cell'];
+
+    if (tableValue !== 0) {
+      fields[i].classList.add(`field-cell--${tableValue}`);
+    }
+  }
+}
+
+updateGameFields();
+
+function appearingOneRandomCell() {
+  const fieldIndex = getRandonCell();
+  const rowIndex = Math.floor(fieldIndex / 4);
+  const colIndex = fieldIndex - (rowIndex * 4);
+
+  const firstRandomNum = Math.floor(Math.random() * 10);
+
+  if (firstRandomNum === 4) {
+    table[rowIndex][colIndex] = 4;
+  } else {
+    table[rowIndex][colIndex] = 2;
+  }
+}
+
+function restart() {
+  table[0] = [0, 0, 0, 0];
+  table[1] = [0, 0, 0, 0];
+  table[2] = [0, 0, 0, 0];
+  table[3] = [0, 0, 0, 0];
+
+  scoreValue = 0;
+  updateScore();
+  messageStart();
+}
+
+button.addEventListener('click', () => {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'restart';
+
+    appearingOneRandomCell();
+    updateGameFields();
+    appearingOneRandomCell();
+    updateGameFields();
+    // console.log(table);
+  } else {
+    button.classList.remove('restart');
+    button.classList.add('start');
+    button.textContent = 'Start';
+
+    restart();
+    updateGameFields();
+    // console.log(table);
+  }
+});
+
+function getRandonCell() {
+  const arr = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
+
+  for (let i = 15; i >= 0; i--) {
```

Comment:
lloks like redundant to make such hardcoded arr.
better to iterate through table, filter empty cells
and than randomly choose one,
can be something like that
```
function getRandonCell() {
const emptyCells = [];

  table.forEach((row, rowIndex) => {
    row.forEach((cell, columnIndex) => {
      if (cell === 0) {
        emptyCells.push([rowIndex, columnIndex]);
      }
    });
  });

  return emptyCells[Math.floor(Math.random() * emptyCells.length)];
}